src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/ConjugatedPiSystemFragmenter.java

```diff
@@ -186,21 +188,22 @@ public List<IAtomContainer> fragmentMolecule(IAtomContainer aMolecule) throws Nu
 
         //<editor-fold desc="Molecule Cloning, Property and Arrays Set">
         IAtomContainer tmpClone = aMolecule.clone();
-        IAtom[] tmpAtomArray = new IAtom[tmpClone.getAtomCount()];
-        IBond[] tmpBondArray = new IBond[tmpClone.getBondCount()];
+        this.clearCache();
+        this.atomArray = new IAtom[tmpClone.getAtomCount()];
+        this.bondArray = new IBond[tmpClone.getBondCount()];
         int tmpCPSFAtomIndex = 0;
         int tmpCPSFBondIndex = 0;
         for (IAtom tmpAtom: tmpClone.atoms()) {
             if (tmpAtom != null) {
                 tmpAtom.setProperty(ConjugatedPiSystemFragmenter.INTERNAL_CPSF_ATOM_INDEX_PROPERTY_KEY, tmpCPSFAtomIndex);
-                tmpAtomArray[tmpCPSFAtomIndex] = tmpAtom;
+                atomArray[tmpCPSFAtomIndex] = tmpAtom;
```

Comment:
Remember to use the "this" statement when using class variables. Here and down below.

Reply:
Of course, mishap on my side.